app/src/main/java/com/sun/premierleague/ui/team/TeamFragment.kt

```diff
@@ -0,0 +1,149 @@
+package com.sun.premierleague.ui.team
+
+import android.view.View
+import androidx.core.os.bundleOf
+import com.google.android.material.appbar.AppBarLayout
+import com.google.android.material.appbar.CollapsingToolbarLayout
+import com.sun.premierleague.R
+import com.sun.premierleague.base.BaseFragment
+import com.sun.premierleague.data.model.Coache
+import com.sun.premierleague.data.model.Player
+import com.sun.premierleague.data.model.TeamItem
+import com.sun.premierleague.utils.RepositoryFactory
+import com.sun.premierleague.utils.loadImage
+import com.sun.premierleague.utils.showToast
+import kotlinx.android.synthetic.main.fragment_detail_team.*
+
+class TeamFragment : BaseFragment(), TeamContact.View {
+
+    private var presenter: TeamPresenter? = null
+    private val adapterGoalkeeper = TeamAdapter()
+    private val adapterDefender = TeamAdapter()
+    private val adapterMidfielder = TeamAdapter()
+    private val adapterForwarder = TeamAdapter()
+
+    override val layoutResource get() = R.layout.fragment_detail_team
+
+    override fun initData() {
+        initAdapter()
+        initPresenter()
+        initCollapsingToolbar()
+        arguments?.getString(BUNDLE_ID_TEAM)?.let {
+            presenter?.getTeamInformation(it)
+        }
+    }
+
+    private fun initAdapter() {
+        recyclerGoalkeeper.adapter = adapterGoalkeeper
+        recyclerPlayerDefend.adapter = adapterDefender
+        recyclerPlayerMidfield.adapter = adapterMidfielder
+        recyclerPlayerForward.adapter = adapterForwarder
+    }
+
+    private fun initPresenter() {
+        val context = context ?: return
+        val repository = RepositoryFactory.getRepository(context)
+        presenter = TeamPresenter(this, repository)
+    }
+
+    private fun initCollapsingToolbar() {
+        val collapsingToolbarLayout =
+            collapsingToolbarTeam as CollapsingToolbarLayout
+        val appBarLayout = appBarTeam as AppBarLayout
+        var isShow = true
+        var scrollRange = -1
+        appBarLayout.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { barLayout, verticalOffset ->
+            if (scrollRange == -1) {
+                scrollRange = barLayout?.totalScrollRange!!
+            }
+            if (scrollRange + verticalOffset == 0) {
+                collapsingToolbarLayout.title = textDetailTeam.text
+                isShow = true
+            } else if (isShow) {
+                collapsingToolbarLayout.title = " "
+                isShow = false
+            }
+        })
+    }
+
+    override fun showTeam(data: List<TeamItem>) {
+        textDetailTeam.text = data.first().teamName
+        imageDetailTeam.loadImage(data.first().teamBadge)
+    }
+
+    override fun showCoach(coache: List<Coache>) {
+        if (coache.isEmpty()) {
+            view?.context?.showToast(R.string.err_no_coach)
+            return
+        }
+        val coachOne = coache.first()
+        textNameCoachOne.text = coachOne.coachName
```

Comment:
```suggestion
        textNameCoachOne.text = coachOne.name
```
tương tự với các props khác